### DISCUSSION TITLE
Remove language selection flags to solve some problems

### DISCUSSION THREAD

**Discussion by SumiHeart** [multitheftauto/mtasa-blue#2163] [General] [answered]
In the language selection interface, some specific language flags may cause some players disagree, so I hope that the language flags can be remove, or the flags of all languages can be removed.

**What is the problem?**
As ccw said:
> Flags make the language selection menu look pretty, but there are problems with flags other than Taiwan.

Also as described on the related website:
> Remember Flags are not Languages

Some scripters think that it has nothing to do with them, but this is limited to them. Maybe scripters in other relevant countries want host servers today, but their players may care about this matter, but they just didn't express their opinions.

**What to do next:**
Face reality, more and most modern games or websites have long abandoned flags and only use text or simple icons instead.

Maybe flags make the interface look better, but I hope you can try switching places with the people who care about it.

**Comment by Dutchman101** [comment]
Note that this issue builds upon a discussion in PR #2151



> that the language flags can be remove

In my opinion we should be speaking of "flag" instead of "flags", namely we could remove a flag display from zh-TW (taiwanese) language in order to solve any political disagreements on which flag to use.

"flags" implies that you want both the Chinese flag (for zh-CN, chinese language selection) and any of such use for zh-CN removed. This seems unreasonable to me, as China is also a country with a significant amount of MTA players, and they do like to use the feature of language selection with flags. Bottom line, the use of the Chinese flag for mainland China has nothing to do with the entire Taiwanese debate, and therefore we should leave the Chinese flag on zh-CN alone. It is a flag that belongs to that country, which unlike for Taiwan isn't subjected to debate.

Now on to something else that annoys me about this matter..
> or the flags of all languages can be removed for the consistent and neat layout

The easily accessible language selection menu is useful and a well-appreciated feature for MTA users around the world. Having flags in that menu (right from the landing page) adds significantly to the user experience and usability of this feature.

Now are we really about to reduce or eliminate that because of some long toes in the world, for people being offended due to something political that pertains to only 1 or 2 countries? I don't think so. We can resolve the issue at hand (related to Taiwan) by doing something like removing the Chinese flag from Taiwan, if we even feel persuaded to do so at all, but to remove an entire feature that serves many countries in the world goes too far in my eyes.

The issue at hand is already nitpicky, so if we were to make Taiwan flagless to satisfy people with long toes, then there's nothing else to warrant 'balancing for a consistent and neat layout' (the presence of 1 country without a flag, in the languages overview) with such an ultimate step as removing the entire languages menu. Then it's give or take, just to do nothing else to it. The issue isn't worth the measure, so as to say. If some people really can't live with the languages menu then being less neat, we can always opt to remove the Taiwanese language from the overview, if really needed.

**Comment by SumiHeart** [comment]
> as China is also a country with a significant amount of MTA players, and they do like to use the feature of language selection with flags.

Just ask for this content: 
Do you know how many Chinese people still playing MTASA? and how do you know about that? 

thank you.

As I know,all mta forum in China,almost people left,just like a deserted city. 
yes maybe they played before,but please,turn one’s back on the fact,better way to make mta more better,why not?

**Comment by ccw808** [comment]
Flags make the language selection menu look pretty, but there are problems with flags other than Taiwan.
e.g. Players in Ireland, Australia etc might not like having to select the British flag. The Arabic flag is represented by KSA but Arabic is spoken in many other countries. Similar issues with Spanish, French, German etc
https://www.google.com/search?q=localizarion+flags

**Comment by xLive** [comment]
> 
> 
> Flags make the language selection menu look pretty, but there are problems with flags other than Taiwan.
> e.g. Players in Ireland, Australia etc might not like having to select the British flag. The Arabic flag is represented by KSA but Arabic is spoken in many other countries. Similar issues with Spanish, French, German etc
> https://www.google.com/search?q=localizarion+flags

The Arabic language is not represented by any country in MTA

![image](https://user-images.githubusercontent.com/30420446/112715717-0b236c00-8ef3-11eb-89cd-06795bc6320a.png)
Arabic is called "the language of ض (Dhad)" because it's the only language that has that letter.

**Comment by Disinterpreter** [comment]
> 
> 
> Flags make the language selection menu look pretty, but there are problems with flags other than Taiwan.
> e.g. Players in Ireland, Australia etc might not like having to select the British flag. The Arabic flag is represented by KSA but Arabic is spoken in many other countries. Similar issues with Spanish, French, German etc
> https://www.google.com/search?q=localizarion+flags

But we have different translations for these languages. 🤔 The flags can be different too.

The same if we would have American English and British English. I saw this difference few times, they had different flags.

![image](https://user-images.githubusercontent.com/3816155/112716122-78d09780-8ef5-11eb-8174-f4d5e01a4b0a.png)

**Comment by patrikjuvonen** [comment]
Show chinese flag on taiwan for chinese (china) language users 😆

**Comment by AlexTMjugador** [comment]
> But we have different translations for these languages. thinking The flags can be different too.
> 
> The same if we would have American English and British English. I saw this difference few times, they had different flags.
> 
> ![image](https://user-images.githubusercontent.com/3816155/112716122-78d09780-8ef5-11eb-8174-f4d5e01a4b0a.png)

The problem with that idea is that MTA has nowhere near the same amount of regional variants of a language than countries that speak such language in many cases. For instance, MTA has a single Spanish localization, and it's represented by the flag of Spain, while Spanish is also spoken in a bunch of countries in Latin America and even in the United States to some extent. The China/Taiwan situation at hand just makes this even worse, as those countries do not formally agree on their political status.

> Do you know how many Chinese people still playing MTASA? and how do you know about that?

MTA collects player stats that, IIRC, count how many players come from what region. But such detailed information is not publicly accessible, though.

I agree with @ccw808 point. Flags make the language selection menu prettier, but I think that, objectively, the best way to represent languages that doesn't favor one country over another or takes part in any political debate is just to get rid of them, and that approach [seems to be accepted](https://ux.stackexchange.com/a/11894) in the GUI design community.

Edit: perhaps a relatively easy solution to these problems that wouldn't require to change the layout of the language selection menu is to substitute all the flags with generic text icons that contain the ISO language code, like this (image and idea brutally copied from this [Stack Exchange answer](https://ux.stackexchange.com/a/2473)):

![Generic language text icons](https://i.stack.imgur.com/JC89R.png)

This way the menu would still look somewhat pretty, while avoiding these kind of issues altogether.

**Comment by PlatinMTA** [comment]
Is this really a problem outside the thing with China and Taiwan? That issue is mostly political more than anything else.

If we really wanted to make everyone feel at home with the languagues (even tho I had never seen a single spanish speaking user be upset because the lang flag is the Spanish one instead of \<insert country here\>) we could do the dafont.com approach, which is:

![image](https://user-images.githubusercontent.com/23291573/112732966-f0b1b880-8f1b-11eb-8f1b-2a4d137449ee.png)

Use the IP location and change the flag depending on your current country. For example: 

1. I am from Uruguay
2. Do a geoIP check
3. See that I'm from an Spanish speaking country
4. See that is Uruguay
5. Change the Spanish flag for the Uruguayan flag
6. There are no more flag problems

No one is ever going to be mad because their country flag is the one that is representing their first languague, instead they would find it neat that the game recognices their location and changes their flag so they feel more at home.

If think this isn't really an issue. It's usually common to use the British flag for english because it's were the english languague was form, and you can go back in the history of that nation until you find the birth of that languague. Same for Spain, France, Italy, Portugal, is just a nit-picking issue. However this is not so nit-picking with Taiwan and China , because it has real implications with actual politics, which MTA shouldn't take an stance on. I think that the solution dafont.com has for this issue is the ideal for solving this flag dispute, at least in the chinese speaking community.

**Comment by LosFaul** [comment]
specially because of political reasons, flags shouldn't be used for language selection at all

**Comment by Disinterpreter** [comment]
> 
> 
> Is this really a problem outside the thing with China and Taiwan? That issue is mostly political more than anything else.
> 
> If we really wanted to make everyone feel at home with the languagues (even tho I had never seen a single spanish speaking user be upset because the lang flag is the Spanish one instead of <insert country here>) we could do the dafont.com approach, which is:
> 
> ![image](https://user-images.githubusercontent.com/23291573/112732966-f0b1b880-8f1b-11eb-8f1b-2a4d137449ee.png)
> 
> Use the IP location and change the flag depending on your current country. For example:
> 
>     1. I am from Uruguay
> 
>     2. Do a geoIP check
> 
>     3. See that I'm from an Spanish speaking country
> 
>     4. See that is Uruguay
> 
>     5. Change the Spanish flag for the Uruguayan flag
> 
>     6. There are no more flag problems
> 
> 
> No one is ever going to be mad because their country flag is the one that is representing their first languague, instead they would find it neat that the game recognices their location and changes their flag so they feel more at home.
> 
> If think this isn't really an issue. It's usually common to use the British flag for english because it's were the english languague was form, and you can go back in the history of that nation until you find the birth of that languague. Same for Spain, France, Italy, Portugal, is just a nit-picking issue. However this is not so nit-picking with Taiwan and China , because it has real implications with actual politics, which MTA shouldn't take an stance on. I think that the solution dafont.com has for this issue is the ideal for solving this flag dispute, at least in the chinese speaking community.

yes, but besides the flags we have different translation for Taiwan and China 🤔

**Comment by PlatinMTA** [comment]
> yes, but besides the flags we have different translation for Taiwan and China 🤔

Show only the translation for Taiwan for Taiwan users, show only the translation for China to Chinese users.* As Dutchman said, the flags, apart from being neat, are helpful for users arround the world. It would suck to have that feature removed because of politics.

\* If this cant be done (i know that some dialects cant really understand each other), then patrik already gave a solution. Even if it was joking it would certanly work.

> Show chinese flag on taiwan for chinese (china) language users 😆

**Comment by Dutchman101** [comment]
I agree with Platin, besides the china-taiwan issue (which in my eyes is actually an issue, yes) everything else is probably worthless to change, as who would get offended because their language is represented by the flag of a bigger or origin country using that language (e.g KSA for Arabic, France for French, etcetera). We haven't gotten any of such complaints from users before, so you can deduce that barely anyone to no one is getting offended.. all we would know is that they care to just easily find their own language, knowing it's probably under the flag of the most known country to speak it.

And if not so, they can just leave the selection menu on the landing page behind and go into options to select their language without dealing with any flags. It's an use or don't use scenario. Besides, that especially for people that have trouble understanding the menu language right after installing MTA, a flags menu on the landing page is intuitive to get them on the right track, to avoid any pain or failure of navigating all the way to Settings

**Comment by AlexTMjugador** [comment]
@PlatinMTA's approach is okay with me, but I find it somewhat complex when we could just opt to not use flags at all, which are not meant to represent languages anyway, and aligns with current GUI design practices. All of this without changing a single line of code.

I really wonder if the menu navigability and user expectations are going to be impacted negatively enough to justify using flags instead of another visual representation. Do we have any hard data about this? In this issue it seems that there's not a clear majority or consensus, and maybe such data can allow us to make a better decision.

**Comment by LosFaul** [comment]
first of all a lot websites and software offer this two types of chinese:
- Chinese (Simplified)
- Chinese (Traditional)


also lot of websites/software make a scrollable list without flags
for example:
- ...
- English (UK)
- English (US)
- French (Français)
- German (Deutsch)
- Spanish (Español)
- ...


geoIP can also go sometimes wrong

**Comment by Dutchman101** [comment]
> I really wonder if the menu navigability and user expectations are going to be impacted negatively enough to justify using flags instead of another visual representation

It's rather the other way around - having received no complaints about the general use of flags for languages in the past.. would anything justify removing the entire feature? We risk that it will reduce user experience & ease of configuring an understandable language for those that would be instantly lost in menu's they cannot understand, for nothing that is evidently 'real' yet (as in: a lack of complaints, and a lack of evidence that having to select your own language under a flag covered by another country really offends a part of our users)

It's an unnecesary risk, as far i can see from what we have here. Also not to forget that in the few years we have the flag selector, people may have gotten used to it.

**Comment by AlexTMjugador** [comment]
> It's rather the other way around - having received no complaints about the general use of flags for languages in the past.. would anything justify removing the entire feature?

If we agree that "removing the entire feature" means "changing the icons from flags to another visual representation", then I can see some reasons for the change:

- The flag of a country may change over time, especially for politically unstable countries. Any such change would require MTA to update the affected flags.
- As @xLive said, the arabic language is already not represented by a flag. Extending this to the rest of languages would only make the language selection menu more consistent.
- It's arguably confusing and ill-posed to use flags to represent languages. While every country has a single flag and each flag belongs to a single country (in general), several languages may be spoken in a country, and a language may span several countries. Moreover, for instance, showing a Spain flag along the Spanish language may be interpreted to mean the Spanish variant spoken in Spain, which uses _ordenador_ instead of _computadora_ to mean _computer_, while AFAIK the MTA Spanish localization tries to be as neutral as possible. And Spanish is not the only language where this happens.
- If, despite the previous point, we try our best to choose a flag to represent a language, we won't be able to do so in a way that's politically correct for everyone. Sure, the origins of every language can be traced back to some country, but using the flag of the country of origin of a language can be interpreted by some people as a way to reaffirm colonialism and/or the perceived superiority of that country. On the other hand, as the Taiwan/China issue at hand shows, a given country might not even have an universally recognized flag. And mixing flags together doesn't work well with more than a few countries.
- Not being politically correct for everyone in a matter like this may, depending on personal opinions and political developments outside of MTA control, make players feel uncomfortable, and even clash with local laws, which ultimately are not good things for MTA.
- Although it's true there were no complaints in the past about this, the current issue definitely triggered some healthy discussion about the _status quo_, which indeed can, in my opinion, be improved. I think this justifies putting some thought and/or action in the matter.
- Assuming that users are not extremely used to seeing flags in that menu already, which they shouldn't, because by no means using flags to represent languages is universal, an alternative visual representation of the language that is as widely recognized as possible (its ISO code, some characteristic element of that language, like the letter _ñ_ for Spanish, etc.) would provide equivalent user experience and ease of navigation. Windows already represents the selected language in the taskbar with ISO codes, for example.

Ultimately, these changes boils down to replacing some PNG files, which can be done by anyone at any time, without requiring any coding knowledge, so I don't think they have to slow down or distract main MTA developers from "more important" topics.

**Reply by SumiHeart** [reply]
I really agree and like that you said,actually I just want to find a best way to solve it,not really have to remove all flags,I thought there is a chance and a way to solve it,but I was wrong,all MTA team members disagree me.

I don't want to emphasized it's a political problem,and I don't think it either,I just hope there is a way,more peaceful,but when team member said they can remove my language,i feel so terrible.

They just bring the problem to "political" but not compromise/middle ground...

They all emphasized political correctness,as Lpsd said:

> First time seeing somebody raise this issue.

This is enough to prove that they are first time to see it,they don't have any experience to solve this kind of problem.

**Reply by Lpsd** [reply]
Because this is the first time an issue is raised, we're ill-equipped to deal with it? Please, don't post complete nonsense.

Yes, we're all aware that countries have history, and people from different sides don't agree on the history. Let's not transfer this mindset into MTA for absolutely no reason apart from the sake of argument (honestly, this whole discussion seems like people are arguing just to argue).

Personally I'd just leave it how it is, however the **only** other option is to remove the flags entirely (or have an option to hide them). We clearly cannot satisfy all users by simply changing the flags.

If you believe there is a solution besides removing the flags, then provide it. So far all I hear is "woe is me" and "mta team don't know how to fix this" while you provide absolutely no alternate solution to the ones already offered by us.

If you have a point, make it.

**Reply by Lpsd** [reply]
> all MTA team members disagree me.

At which point did you forget that most of this discussion stems from your opinion or political views? Yes, people are going to disagree.

> They just bring the problem to "political" but not compromise/middle ground...

We've provided middle ground by offering by having an option to remove the flags (or just removing them entirely). Please provide your solution.

> but when team member said they can remove my language,i feel so terrible.

Please confirm what you mean here. Who said this, when and where?

We are not "against" people bringing up issues like this, that's fine. However, don't get offended or upset when your ideal outcome to the problem doesn't become reality. 

If you truly believe no middle ground has attempted to be made throughout this whole discussion, I would advise you to come back with a clear mind and re-read the entire conversation.

**Reply by SumiHeart** [reply]
okay Lpsd,clam down please.

> > but when team member said they can remove my language,i feel so terrible.
> 
> Please confirm what you mean here. Who said this, when and where?

here
> we can always opt to remove the Taiwanese language from the overview, if really needed.


and about this
> So far all I hear is "woe is me" and "mta team don't know how to fix this" while you provide absolutely no alternate solution.

I think ccw808 is giving a good solution,I think it's enough,although other team members may disagree,I think I don't have to provide other point now,because for other team members,they may think what I said is for political.

Thank you Lpsd. I am sorry if I had offended you.

**Reply by Lpsd** [reply]
You said:

> I just want to find a best way to solve it,not really have to remove all flags

Which is why I asked you to provide YOUR solution. And now:

> I think ccw808 is giving a good solution,I think it's enough

Now you're also being facetious ("sorry for offending you", "calm down", etc) because what you're saying has no substance - please, what a waste of my time.

**Reply by SumiHeart** [reply]
I'm sorry about waste of your important time.
But you can really ignore me,if you are think about that I'm waste your time,don't need to be so angry and waste time.

**Reply by Lpsd** [reply]
You're wasting my time by making baseless comments such as "don't be so angry" and "calm down" instead of providing answers to genuine questions. 

The only answer you did give actually shot yourself in the foot, because it went against everything you initially said (again, a flimsy debate full of holes because you're arguing just for the sake of arguing at this point).

Quick tip: the "poor me" mentality doesn't help move the discussion along.

I'll say it again - we've provided a couple of options to deal with this issue. If you do not agree with the current solutions, please provide another. 

This will be my last response on this topic.

**Reply by Dutchman101** [reply]
> Let's not transfer this mindset into MTA for absolutely no reason apart from the sake of argument (honestly, this whole discussion seems like people are arguing just to argue)

Unfortunately, it appears that people from the general open-source community, that are interested in, and actively looking for, political discussions/"im offended"/long toes arguments, have entered this discussion. Multiple staff already had to edit multiple comments, to cut out offensive and insulting, hot-headed content.

I agree with you @Lpsd but we should probably realize this and not lose ourselves in the toxicity that's being offered to us.
Btw, on topic, i really like the "flagless country list design" that ccw last suggested.

**Comment by PlatinMTA** [comment]
I think it is kind of pointless to keep arguing about this. You can find almost any reason to be against the flags, and almost every answer to that issue. This just boils down to politics, political correctness and personal preference.

Someone from the MTA Team (Admin) should decide what to do and that's it, I don't think we will ever agree on this topic.

**Comment by Dutchman101** [comment]
Maybe we should handle it on a case-by-case basis, so when someone first gets offended/feels uncomfortable (as i think is nicely worded in AlexTMjugador's post) they can add an issue, referring to this discussion.

The china/taiwan situation is a good example of where it would be warranted.

**Reply by AlexTMjugador** [reply]
To be honest, I think that would be a good compromise/middle ground between all the exposed ideas.

**Comment by sbx320** [comment]
- Removing flags entirely makes it significantly harder to find the desired language, since for most users "flag == language" holds true.
- GeoIP (in any variant) doesn't sound like a good solution to me since it adds significant complexity and can also go very wrong at times.
- I don't think we should take any action just because some Chinese players may be offended by a Taiwanese flag. 
- For all other countries I don't think this is a significant issue that people even care enough about (see also: lack of complaints). 

To me the current solution (using flags) still seems like the best solution, so I'd be in favor of closing this as a wontfix. 

---

> Edit: perhaps a relatively easy solution to these problems that wouldn't require to change the layout of the language selection menu is to substitute all the flags with generic text icons that contain the ISO language code, like this (image and idea brutally copied from this Stack Exchange answer):

Those language codes seem rather pointless, since they don't allow for quickly finding the desired language for most users.

**Comment by LaFantasy** [comment]
Although I'm Italian, I do not support the use of flag as a language.

**Comment by Lpsd** [comment]
First time seeing somebody raise this issue.

This will never be resolved as it's political and largely based on opinion. 

I don't think we start changing everything based on political correctness. MTA should be distancing itself from these kind of topics due to the client assuming our culpability when they are offended due to "wrong decisions" or statements made by MTA on these issues, based on their opinions.

I'd also be in favour of closing this. I'm fairly certain this isn't ruining anyone's life or stopping them from playing MTA (if it is, please seek help).

**Reply by AlexTMjugador** [reply]
> MTA should be distancing itself from these kind of topics due to the client assuming our culpability when they are offended due to "wrong decisions" or statements made by MTA on these issues, based on their opinions.

While I agree with this goal, I think it still feels a bit rude to close issues like this with statements along the lines of "we made that with good will, and if you're offended it's your fault for assuming our culpability", even though they're genuine. It definitely does not show much tact for people that may take it seriously. In my opinion, I think we should at least listen to such complaints when they appear and find a suitable middle ground for each case, as @Dutchman101 proposed.

After all, even if that middle ground involves removing a single flag for a single country, that is still better than doing nothing at all, while minimally impacting any navigability or user experience.

**Reply by Lpsd** [reply]
> even though they're genuine

Genuinity doesn't reflect the issue at hand. We're crossing into political and opinion territory - the only thing genuine is the belief in one's own personal opinions.

I feel like that point has been made clear. What happens when we make a change because of one user - just for another user to complain because he has the opposite opinion? Do we revert the change then?

I have a catch-all solution, add "hide flags" to MTA settings menu. If people do not agree with the current flags displayed, they can simply disable them.

**Reply by AlexTMjugador** [reply]
> I have a catch-all solution, add "hide flags" to MTA settings menu. If people do not agree with the current flags displayed, they can simply disable them.

The problem I can see with that solution is what default value to choose for the setting. On one hand, if you choose to show the flags by default, you risk giving a bad impression to people that feels uncomfortable with the flags. On the other hand, if you hide flags by default, you lose the potential UX benefit of using flags in the first place.

But yeah, I fully agree with your point that "the only thing genuine is the belief in one's own personal opinions", and I appreciate we're talking about this in a civil and constructive manner.

**Comment by jey-banned** [comment]
+1 for closing this issue. It almost sounds like "Why is there the UK flag on English language? I'm from USA and i disagree!" Some popular websites use the UK flag, while others use the US flag. Their owners don't have any problems by doing that

**Comment by patrikjuvonen** [comment]
I have transformed this into a discussion instead of an issue.

**Comment by ccw808** [comment]
Putting aside the issue of politics, I have been thinking about removing the flags for a while because it is so 2010's and also inconsistent with the installer language selection which does not use flags. MTA in-game language selection didn't use to have flags, and apart from making it look pretty, adding flags doesn't make it easier to select the desired language.
Here is a mock up of the language menu without flags and it's just as quick to find the language you want, I think
![mta-lang-menu](https://user-images.githubusercontent.com/123884/112997556-23022680-9165-11eb-8f93-611ef7a6d5c7.jpg)

**Reply by Disinterpreter** [reply]
Yes, it looks really good and solve all problems with this question.

**Reply by AlexTMjugador** [reply]
I also really like this menu. It looks so modern, clean and it's free of any political elements.

**Reply by Disinterpreter** [reply]
Well... I think it is a solution. We have to make an issue and the discussion can be solved.

**Comment by ErasureCode** [comment]
In Mexico, I used to own several servers, but i received complaints from some players that they thought that the Spanish flag could not be used in Spanish because we also use it. It would be very stupid to go back to history to confirm the flag of the language, because history has many painful costs.

**Reply by PlatinMTA** [reply]
If we are using the flags in the context to make it easy to navigate a languague selector, then I don't think it's stupid nor arrogant to say that we should use common flags were we know that language probably originated from and, specially, is were the name of the language actually came from.

Will you understand right away which language are you selecting if you saw, for example, the Bolivian Flag? Sadly most users won't understand it. Users from Bolivia would certanly do however, but not all of the Spanish Speaking users. You know what connects all of us Spanish Speakings countries of America? That we were subject to the Spanish Crown, and therefore we inherit the Spanish Language.

That's all the flag needs to represent. It does not need to represent the painful past, the humilliation and erradication of our people, the constat wars or the riches that were stolen, it just needs to represent the language, and I think that's all we are discussing here.

The only thing that is different with the Taiwan/China case is that is current history, it is happening right now, and has been hapening for some decades. Our ancestors were the ones who dealed with the Spanish Conquistadors and Portuguese Colonizers, while now we don't have to and we would probably never need to fight for the freedom of our nations ever again. Could we say the same about the Taiwanese?

Having that aside I'll still argue that this Discussion won't go anywhere, there is too much political and personal preference on this topic, which I think is clearly visible, and it will create a biased and pointless discussion, both ways.

**Comment by MoonWindow** [comment]
Point of view of a neutral player, I really don’t need a flag and I can simply   identify my language, because in all languages, I can only know my   language.     It’s very simple. Spaniards can’t read Chinese, so I can’t be   confused.          I am a player, and the flag has not really optimized the interface. If I   were Panamanian, would I be stuck thinking like you just like EVER GIVEN?     No, because I use Spanish, but I am a Panamanian, and I can’t find my flag,   so I look at the language text to found my language.          Poor guys..

**Comment by Dezash** [comment]
Politics aside, flags were never meant to represent languages. Modern software doesn't use them and personally, I've never had trouble picking my language nor have I ever heard anyone complain that there are no flags. In fact, I've noticed some funny trends in my server where Russian players would sometimes end up selecting Dutch as their language and I have a hunch that they may have been confusing their flag. If you want to improve UX, add a search box instead.

**Reply by AlexTMjugador** [reply]
I find the search box to be a great idea to quickly find your desired language, should the language list become so big that finding your language is hard.

**Comment by patrikjuvonen** [comment, marked as answer]
Resolved by https://github.com/multitheftauto/mtasa-blue/commit/9213854c2534b8cb3e340f2101a5ffee6c667daa
